jest.config.js

```diff
@@ -31,7 +31,7 @@ module.exports = {
 	],
 	moduleNameMapper: {
 		'@wordpress\\/(blocks|editor)$': '<rootDir>/gutenberg/$1',
-		'@wordpress\\/(data|element|deprecated)$': '<rootDir>/gutenberg/packages/$1',
+		'@wordpress\\/(data|element|deprecated)$': '<rootDir>/gutenberg/packages/$1/src/index',
```

Comment:
I believe this is the key to this PR - could you explain how this would be benefitial, with an example, maybe? Otherwise I can only assume you're going to be using this elsewhere (it's not apparent from this PR itself, what the value it may add is)

Reply:
Also, this is the jest configuration, but do we have a place where it's made clear and apparent that the main configuration uses the source files?

Reply:
> I believe this is the key to this PR

Not quite key per se, for 2 reasons:
1. It modifies how the packages are resolved (overwritten in this case) for the _tests_ (Jest), not the app code
2. It's basically a workaround because Jest resolves the entry point of a package by reading the `main` field in the `package.json`. For example, for the `element` package the entry point is the *built* file. [See here on the GB repo](https://github.com/WordPress/gutenberg/blob/master/packages/element/package.json#L20-L22). In an ideal world, the Jest tests would use the same entrypoint as the RN app so we wouldn't need to configure Jest separately.

A question one could raise here is: *why do we even want to use the package _source_ files instead of the already transpiled ones?* The answer is that we prolly want to do that for the near future so we can iterate fast on the packages themselves. For example, with the current PR, one can modify the package `src/index.js` and have the change be picked up immediately by the RN app. With the built files, one would need to manually go through `npm install` to rebuild the packages before using the change.

I expect that in the not-so-distant future, we will revert back to using the built files after the project matures enough and we only introduce changes to the packages rarely.

> do we have a place where it's made clear and apparent that the main configuration uses the source files?

Yes, it's the `react-native` entry point, [here for example on the GB repo](https://github.com/WordPress/gutenberg/blob/master/packages/element/package.json#L22).

Reply:
> Not quite key per se, for 2 reasons:

Right,  I realized it was just for jest in my second comment :) - thanks for pointing to the `main`,  `module` and `react-native` keys in `package.json`, these values made sense 👍

This paragraph here is 🏅 :

> A question one could raise here is: why do we even want to use the package source files instead of the already transpiled ones? The answer is that we prolly want to do that for the near future so we can iterate fast on the packages themselves. For example, with the current PR, one can modify the package src/index.js and have the change be picked up immediately by the RN app. With the built files, one would need to manually go through npm install to rebuild the packages before using the change.

This is probably what I'd find useful the most in the PR description to understand the purpose of the PR itself - maybe a suggestion for future PRs 🙇 (thank you very much for the clarification)